src/dmd/traits.d

```diff
@@ -412,6 +413,31 @@ extern (C++) Expression pointerBitmap(TraitsExp e)
     return ale;
 }
 
+/**
+* __traits(getCallerSource, symbol) => get symbol source code (string[T.length] or string)
+*/
+extern (C++) Expression traitsGetSource(TraitsExp e)
+{
+    if (!e.args || e.args.dim < 1|| e.args.dim > 2)
+    {
+        error(e.loc, "TODO");
+        return new ErrorExp();
+    }
+    auto ret = new ArgStringInitExp(e.loc);
```

Comment:
This is allocated here, but is not initialized or used until the end of the function. It should be allocated there. Also, convention is to use `result` as the return value, not `ret`.